src/main/java/stream/flarebot/flarebot/util/general/GeneralUtils.java

```diff
@@ -191,194 +401,38 @@ public static AudioItem resolveItem(Player player, String input) throws IllegalA
-        return getRole(s, guildId, null);
-    }
-
-    public static Role getRole(String s, String guildId, TextChannel channel) {
-        Guild guild = Getters.getGuildById(guildId);
-        Role role = guild.getRoles().stream()
-                .filter(r -> r.getName().equalsIgnoreCase(s))
-                .findFirst().orElse(null);
-        if (role != null) return role;
-        try {
-            role = guild.getRoleById(Long.parseLong(s.replaceAll("[^0-9]", "")));
-            if (role != null) return role;
-        } catch (NumberFormatException | NullPointerException ignored) {
-        }
-        if (channel != null) {
-            if (guild.getRolesByName(s, true).isEmpty()) {
-                String closest = null;
-                int distance = LEVENSHTEIN_DISTANCE;
-                for (Role role1 : guild.getRoles().stream().filter(role1 -> FlareBotManager.instance().getGuild(guildId).getSelfAssignRoles()
-                        .contains(role1.getId())).collect(Collectors.toList())) {
-                    int currentDistance = StringUtils.getLevenshteinDistance(role1.getName(), s);
-                    if (currentDistance < distance) {
-                        distance = currentDistance;
-                        closest = role1.getName();
-                    }
-                }
-                MessageUtils.sendErrorMessage("That role does not exist! "
-                        + (closest != null ? "Maybe you mean `" + closest + "`" : ""), channel);
-                return null;
-            } else {
-                return guild.getRolesByName(s, true).get(0);
-            }
-        }
-        return null;
-    }
-
-    public static TextChannel getChannel(String arg) {
-        return getChannel(arg, null);
-    }
-
-    public static TextChannel getChannel(String channelArg, GuildWrapper wrapper) {
-        try {
-            long channelId = Long.parseLong(channelArg.replaceAll("[^0-9]", ""));
-            return wrapper != null ? wrapper.getGuild().getTextChannelById(channelId) : Getters.getChannelById(channelId);
-        } catch (NumberFormatException e) {
-            if (wrapper != null) {
-                List<TextChannel> tcs = wrapper.getGuild().getTextChannelsByName(channelArg, true);
-                if (!tcs.isEmpty()) {
-                    return tcs.get(0);
+    /**
+     * Checks if a perm is valid.
+     *
+     * @param perm The perm to check.
+     * @return if it's valid.
+     */
+    public static boolean validPerm(String perm) {
+        if (perm.equals("*") || perm.equals("flarebot.*")) return true;
+        if (perm.startsWith("flarebot.") && perm.split("\\.").length >= 2) {
+            perm = perm.substring(perm.indexOf(".") + 1);
+            String command = perm.split("\\.")[0];
+            for (Command c : FlareBot.instance().getCommandManager().getCommands()) {
+                if (c.getCommand().equalsIgnoreCase(command) && c.getType() != CommandType.SECRET) {
```

Comment:
Also make this internal, we should probably change getCommands to exclude them automatically and have a getAllCommands if we ever need it.

Reply:
(╯°□°）╯︵ ┻━┻